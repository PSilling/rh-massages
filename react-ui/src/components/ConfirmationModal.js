// react imports
import React, { Component } from 'react';

// component imports
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class ConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <ModalActions
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default ConfirmationModal
