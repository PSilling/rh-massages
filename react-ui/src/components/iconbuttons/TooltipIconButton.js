// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * Labeless icon only button with an attached tooltip.
 */
class TooltipIconButton extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  render() {
    const { disabled, icon, ...rest } = this.props;
    return (
      <span className="no-print">
        <Button {...rest} id={this.tooltipTarget} outline disabled={this.props.disabled}>
          <span className={`fas fa-${this.props.icon}`} />
        </Button>
        {this.props.tooltip !== "" && !this.props.disabled && (
          <Tooltip
            isOpen={this.state.tooltipActive}
            target={this.tooltipTarget}
            toggle={this.toggleTooltip}
            trigger="hover"
          >
            {this.props.tooltip}
          </Tooltip>
        )}
      </span>
    );
  }
}

TooltipIconButton.propTypes = {
  /** icon to be shown instead of a normal label */
  icon: PropTypes.string.isRequired,
  /** function to be called on button click */
  onClick: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button size class */
  size: PropTypes.string,
  /** button component style */
  style: PropTypes.shape({
    border: PropTypes.string
  }),
  /** tooltip to be shown on button hover */
  tooltip: PropTypes.string
};

TooltipIconButton.defaultProps = {
  disabled: false,
  size: "sm",
  style: { border: "0px solid transparent" },
  tooltip: ""
};

export default TooltipIconButton;
