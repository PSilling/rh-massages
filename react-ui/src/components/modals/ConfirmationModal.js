// react imports
import React, { Component } from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";

// module imports
import { ModalContainer, ModalDialog } from "react-modal-dialog";

// component imports
import ModalActions from "../buttons/ModalActions";

// util imports
import _t from "../../util/Translations";

/**
 * Modal with a simple action confirmation message and ModalActions.
 */
class ConfirmationModal extends Component {
  handleKeyPress = event => {
    if (event.charCode === 13 && document.activeElement === ReactDOM.findDOMNode(this.modal)) {
      this.props.onConfirm();
    }
  };

  renderInsides = () => (
    <div>
      <h3>{_t.translate("Action confirmation")}</h3>
      <p>{this.props.message}</p>
      <ModalActions onProceed={this.props.onConfirm} onClose={this.props.onClose} autoFocus>
        {this.props.children}
      </ModalActions>
    </div>
  );

  render() {
    return this.props.withPortal ? (
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog
          onClose={this.props.onClose}
          width="40%"
          style={{ outline: "none" }}
          tabIndex="-1"
          onKeyPress={this.handleKeyPress}
          ref={dialog => {
            this.modalDialog = dialog;
          }}
        >
          {this.renderInsides()}
        </ModalDialog>
      </ModalContainer>
    ) : (
      this.renderInsides()
    );
  }
}

ConfirmationModal.propTypes = {
  /** children supplied to ModalActions */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** message in the Modal */
  message: PropTypes.string.isRequired,
  /** callback function triggered on Modal close */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onConfirm: PropTypes.func.isRequired,
  /** whether ModalContainer should be used; useful for testing to avoid portals */
  withPortal: PropTypes.bool
};

ConfirmationModal.defaultProps = {
  children: null,
  withPortal: true
};

export default ConfirmationModal;
