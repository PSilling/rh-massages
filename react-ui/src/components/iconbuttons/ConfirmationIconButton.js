// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// component imports
import ConfirmationModal from "../modals/ConfirmationModal";
import TooltipIconButton from "./TooltipIconButton";

// util imports
import _t from "../../util/Translations";

/**
 * TooltipIconButton with an action ConfirmationModal.
 */
class ConfirmationIconButton extends Component {
  state = { active: false };

  handleToggle = () => {
    this.setState(prevState => ({ active: !prevState.active }));
  };

  render() {
    const { onConfirm, dialogMessage, ...rest } = this.props;
    return (
      <span>
        <TooltipIconButton {...rest} onClick={this.handleToggle} />
        {this.state.active && (
          <ConfirmationModal
            message={this.props.dialogMessage}
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.props.onConfirm();
            }}
          />
        )}
      </span>
    );
  }
}

ConfirmationIconButton.propTypes = {
  /** function to be called on action confirmation */
  onConfirm: PropTypes.func.isRequired,
  /** message disabled in the ConfirmationModal */
  dialogMessage: PropTypes.string
};

ConfirmationIconButton.defaultProps = {
  dialogMessage: _t.translate("Are you sure? This action cannot be reverted.")
};

export default ConfirmationIconButton;
