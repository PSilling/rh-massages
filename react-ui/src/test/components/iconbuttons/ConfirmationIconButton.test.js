// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import ConfirmationIconButton from "../../../components/iconbuttons/ConfirmationIconButton";
import ConfirmationModal from "../../../components/modals/ConfirmationModal";
import TooltipIconButton from "../../../components/iconbuttons/TooltipIconButton";

test("renders content with correct props", () => {
  const testFunction = jest.fn();
  const wrapper = shallow(<ConfirmationIconButton icon="trash" onConfirm={testFunction} dialogMessage="message" />);
  const button = wrapper.find(TooltipIconButton);

  button.props().onClick();

  expect(wrapper.instance().state.active).toBe(true);
  expect(testFunction).not.toHaveBeenCalled();

  const dialog = wrapper.find(ConfirmationModal);
  dialog.props().onConfirm();

  expect(wrapper.instance().state.active).toBe(false);
  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(dialog.props().message).toEqual("message");
  expect(wrapper).toMatchSnapshot();
});
