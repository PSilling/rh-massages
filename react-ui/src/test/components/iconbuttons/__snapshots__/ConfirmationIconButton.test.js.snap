// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmationIconButton
    dialogMessage="message"
    icon="trash"
    onConfirm={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TooltipIconButton
          disabled={false}
          icon="trash"
          onClick={[Function]}
          size="sm"
          style={
            Object {
              "border": "0px solid transparent",
            }
          }
          tooltip=""
        />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "icon": "trash",
          "onClick": [Function],
          "size": "sm",
          "style": Object {
            "border": "0px solid transparent",
          },
          "tooltip": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TooltipIconButton
            disabled={false}
            icon="trash"
            onClick={[Function]}
            size="sm"
            style={
              Object {
                "border": "0px solid transparent",
              }
            }
            tooltip=""
          />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "icon": "trash",
            "onClick": [Function],
            "size": "sm",
            "style": Object {
              "border": "0px solid transparent",
            },
            "tooltip": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
